js/modules/loki_public_chat_api.js

```diff
@@ -555,17 +582,79 @@ class LokiPublicChannelAPI {
           adnMessage.annotations.length !== 0
         ) {
           const noteValue = adnMessage.annotations[0].value;
-          ({ timestamp, quote } = noteValue);
+          ({ timestamp } = noteValue);
 
-          if (quote) {
-            quote.attachments = [];
+          if (noteValue.quote) {
+            ({ quote } = noteValue);
+            if (quote) {
+              quote.attachments = [];
+            }
           }
 
-          // if user doesn't have a name set, fallback to annotation
-          // pubkeys are already there in v1 (first release)
-          if (!from) {
-            ({ from } = noteValue);
+          if (noteValue.sig) {
+            // try to verify signature
+            const { sig, sigver } = noteValue;
+            const annoCopy = JSON.parse(JSON.stringify(adnMessage.annotations));
+            delete annoCopy[0].value.sig;
+            delete annoCopy[0].value.sigver;
+            const verifyObj = {
+              text: adnMessage.text,
+              version: sigver,
+              annotations: annoCopy,
+            };
+            if (adnMessage.reply_to) {
+              verifyObj.reply_to = adnMessage.reply_to;
+            }
+            const pubKeyBin = StringView.hexToArrayBuffer(
+              adnMessage.user.username
+            );
+            const sigBin = StringView.hexToArrayBuffer(sig);
+            try {
+              await libsignal.Curve.async.verifySignature(
+                pubKeyBin,
+                JSON.stringify(verifyObj),
```

Comment:
Does this work with passing a string?
I've always thought it would only accept an ArrayBuffer

Reply:
yes, I think I tried an arrayBuffer and had problems

Reply:
All good then :)